src/v0/destinations/am/transform.js

```diff
@@ -131,7 +131,7 @@ function createRevenuePayload(message, rawPayload) {
 
 function updateTraitsObject(property, traitsObject, actionKey) {
   const propertyToUpdate = getValueFromMessage(traitsObject, property);
-  if (property && property.length > 0) {
+  if (traitsObject[actionKey] && property && typeof property === 'string' && property.length > 0) {
```

Comment:
Redundant.
```suggestion
  if (traitsObject[actionKey] && property && typeof property === 'string') {
```

Reply:
Empty String passed was what I thought to be the main issue because `actionKey` is always defined to a minimum of a empty Object.

Reply:
`property` condition covers the empty string case also.

Reply:
Then we have to look back to what was the real issue.
On it

Reply:
Earlier, no `property &&` check was there. Now, that's sufficient.

Reply:
Done